### PR TITLE
[DOCS] expand examples on providing mappings for create index and put mapping

### PR DESCRIPTION
**Pull request by javanna**
The create index API and put mappings API docs the for high-level Java REST client didn't have a lot of info on how to provide mappings. This commit adds some examples.

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/IndicesClientDocumentationIT.java [hunk 1]
```diff
@@ -172,24 +176,78 @@ public void testCreateIndex() throws IOException {
...
-                "      }\n" +
-                "    }\n" +
-                "  }\n" +
-                "}", // <2>
-                XContentType.JSON);
-            // end::create-index-request-mappings
+            {
+                // tag::create-index-request-mappings
+                request.mapping("tweet", // <1>
+                        "{\n" +
```
Comment (tlrx) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/IndicesClientDocumentationIT.java]: nit: indentation

File: docs/java-rest/high-level/apis/_index.asciidoc [hunk 2]
```diff
@@ -16,7 +16,8 @@ include-tagged::{doc-tests}/CRUDDocumentationIT.java[index-request-string]
 <4> Document source provided as a `String`
 
 ==== Providing the document source
-The document source can be provided in different ways:
+The document source can be provided in different ways besides the
```
Comment (tlrx) [inline docs/java-rest/high-level/apis/_index.asciidoc]: Nit: besides -> "in addition to" ?

**Review by tlrx — APPROVED** [review]
LGTM, left some minor comments

File: docs/java-rest/high-level/apis/_index.asciidoc [hunk 3]
```diff
@@ -16,7 +16,8 @@ include-tagged::{doc-tests}/CRUDDocumentationIT.java[index-request-string]
...
+`String` example showed above:
```
Comment (olcbean) [inline docs/java-rest/high-level/apis/_index.asciidoc]: nit : showed -> shown

File: docs/java-rest/high-level/apis/create_index.asciidoc [hunk 4]
```diff
@@ -31,6 +31,30 @@ include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[create-index-reque
 <1> The type to define
 <2> The mapping for this type, provided as a JSON string
 
+The mapping source can be provided in different ways besides the
+`String` example showed above:
```
Comment (olcbean) [inline docs/java-rest/high-level/apis/create_index.asciidoc]: as above ;)

File: docs/java-rest/high-level/apis/put_mapping.asciidoc [hunk 5]
```diff
@@ -20,7 +20,32 @@ A description of the fields to create on the mapping; if not defined, the mappin
 --------------------------------------------------
 include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[put-mapping-request-source]
 --------------------------------------------------
-<1> The mapping source
+<1> The mapping source provided as a `String`
+
+==== Providing the mapping source
+The mapping source can be provided in different ways besides the
+`String` example showed above:
```
Comment (olcbean) [inline docs/java-rest/high-level/apis/put_mapping.asciidoc]: one more ;)

**Comment by javanna** [comment]
thanks @tlrx & @olcbean

**Pull request merged**